docs/template_plugin/tests/functional/op_reference/atan.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <vector>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+
+namespace reference_tests {
+namespace {
+
+struct AtanParams {
+    Tensor input;
+    Tensor expected;
+};
+
+struct Builder : ParamsBuilder<AtanParams> {
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, input);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, expected);
+};
+
+class ReferenceAtanLayerTest : public testing::TestWithParam<AtanParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.input.shape, params.input.type);
+        inputData = {params.input.data};
+        refOutData = {params.expected.data};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<AtanParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "shape=" << param.input.shape << "_";
+        result << "type=" << param.input.type;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const Shape& shape, const element::Type& type) {
+        const auto in = std::make_shared<op::Parameter>(type, shape);
+        const auto atan = std::make_shared<op::Atan>(in);
+        return std::make_shared<Function>(NodeVector{atan}, ParameterVector{in});
+    }
+};
+
+TEST_P(ReferenceAtanLayerTest, AtanWithHardcodedRefs) {
+    Exec();
+}
+
+}  // namespace
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_Atan_With_Hardcoded_Refs, ReferenceAtanLayerTest,
+    ::testing::Values(
+        Builder{}
+            .input({{11}, element::f16, std::vector<ngraph::float16>{-4.f, -2.f, -1.f, -0.5f, -0.25f, 0.f, 0.25f, 0.5f, 1.f, 2.f, 4.f}})
+            .expected({{11}, element::f16, std::vector<ngraph::float16>{-1.32581766f,
+                                                                       -1.10714872f,
+                                                                       -0.78539816f,
+                                                                       -0.46364761f,
+                                                                       -0.24497866f,
+                                                                        0.00000000f,
+                                                                        0.24497866f,
+                                                                        0.46364761f,
+                                                                        0.78539816f,
+                                                                        1.10714872f,
+                                                                        1.32581766f}}),
+        Builder{}
```

Comment:
As I see atan evaluate supports boolean type, but we prohibit to create atan with boolean data type. Could you remove support of boolean data type from atan evaluate?

Reply:
Removed boolean type from atan evaluation.